File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> charSet{};

        int res{};

        int l = 0, r = 0;

        while (r < s.length()) {
            while (charSet.contains(s[r])) {
                charSet.erase(s[l]);
                l++;
            }
            charSet.insert(s[r]);

            res = max(res, r-l+1);
            r++;
        }
        return res; 
        
    }
};

File: Data Structures & Algorithms/assign-cookies/submission-0.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:

        i, j = 0, 0

        g.sort()
        s.sort()
        res = 0
        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                res+=1
                i+=1
            j+=1
        return res 
        